### ISSUE TITLE
fix(export from backend to frontend): Be resilient to item serialization failures

### ISSUE THREAD

**Issue by thomass-dev** [probabl-ai/skore#628]
Actually, if one item can't be exported from backend to frontend (for instance https://github.com/probabl-ai/skore/issues/617), all the export phase will failled and the frontend will be empty.

We should skip the failed item, show a warning and go ahead with next items.

**Comment by koaning** [comment]
Bonus points if the warning is explicit about which component caused the error and maybe also points to a document that explains the details on the docs.

**Comment by rouk1** [comment]
We may also display "failing items" in the item tree with some special indicator (maybe leverage tooltips and warning icon) for easier tracking of "broken items". @anasstarfaoui wdyt ?

**Comment by anasstarfaoui** [comment]
I think it could be a nice idea, but maybe we could do that even before, in the UI history @rouk1 ?

**Comment by rouk1** [comment]
> On a technical issue, i can set the content of the item to `<not json-serializable> ` or whatever you want @rouk1 .

I would rather parse an consistent data structure. So we may add an `error` field to the `ProjectItem` interface. It's value could indeed be a message like "item is not json serializable" or maybe something more focused on how the user may fix the issue.

> I think it could be a nice idea, but maybe we could do that even before, in the UI history @rouk1 ?

Your right but we have to handle both cases. An item cant be displayed as a view card and one version of the item cant be viewed as a view card. 
This leads to another question. Should failing items be automatically removed from the history ?
@MarieS-WiMLDS & @sylvaincom wdyt ?

**Comment by MarieS-WiMLDS** [comment]
> Should failing items be automatically removed from the history ?  
 
I don't think so. As a user, I would find it weird and I think I would try to load it again.  
I like better your idea to display these items in the left panel, but with an indicator.

**Comment by anasstarfaoui** [comment]
Okay let's do that then, I'm just not a big fan of mixing broken and valid things inside the sidebar, I feel like it create visual clutter and distract user from using correctly the bar, especially without filters

But let's test it anyway, I will try to design something for this

**Comment by anasstarfaoui** [comment]
Here is a link to what we just said, let me know if it's what you had in mind :)

https://www.figma.com/design/Qzjyxv0lPDU2kKtsH9dcp3/Skore---Workstation---High-fidelity-mock-ups---Q3?node-id=4327-13000&t=OYHGBd70M8Fpw7Mf-4

**Comment by rouk1** [comment]
> Here is a link to what we just said, let me know if it's what you had in mind :)
> 
> https://www.figma.com/design/Qzjyxv0lPDU2kKtsH9dcp3/Skore---Workstation---High-fidelity-mock-ups---Q3?node-id=4327-13000&t=OYHGBd70M8Fpw7Mf-4

Exactly ! 

> I don't think so. As a user, I would find it weird and I think I would try to load it again.
> I like better your idea to display these items in the left panel, but with an indicator.

I meant, as we will soon support multiple version of items. Should we overwrite instead of append if the latest item is in error ?

**Comment by MarieS-WiMLDS** [comment]
> I meant, as we will soon support multiple version of items. Should we overwrite instead of append if the latest item is in error ?  

I think that we should be consistent, whether it's the first or tenth version, and whether the item is displayable or not. If when the item is supported correctly, we append, then it should be the same when we have an error.

**Comment by MarieS-WiMLDS** [comment]
@thomass-dev, can be close this issue, now that we have the pickle fallback?
